web/db/actions/Account.ts

```diff
@@ -99,4 +99,27 @@ async function findAll(
   }
 }
 
-export { findAccount, addAccount, removeAllAccounts, updateAccount, findAll };
+/**
+ * Finds account by search term using regex
+ *
+ * @param searchSubject string in regex
+ * @param session MongoDB session used for transactions
+ * @returns pruned document (without _id, __v) if successful, null otherwise
+ */
+async function searchAccount(
+  searchSubject: string,
+  session?: ClientSession
+): Promise<HydratedDocument<IAccount> | null> {
+  try {
+    const regexTerm = new RegExp(`.*${searchSubject}.*`, 'i');
+    return await Account.findOne(
+      { email: regexTerm },
+      { _id: 0, __v: 0 },
+      { session: session }
+    );
+  } catch (e) {
+    return null;
+  }
+}
```

Comment:
This action needs to return _all_ documents that match the regex pattern. Update tests accordingly. Output should be an array unless there is a database error, in which case return null